### DISCUSSION TITLE
Globally access nextjs req

### DISCUSSION THREAD

**Discussion by mwisner** [vercel/next.js#13497] [Help]
Hi!

I'm wondering if anyone has any decent solutions for accessing the req object from anywhere while running in Node. Similar to how we can use `withRouter` HOC and `useRouter` anywhere in React land... I'd like to be able to do something similar with the req object.

I found this: https://github.com/vercel/next.js/issues/7393. But the accepted solution appears to be one using a custom server. Does anyone have any ideas for how to accomplish this without a custom server?

--- 

More specifically, on the client, I can do this: ``` window.location.host``` to get the host. I need to be able to do something similar on the server. Ideally, if I could write some sort of function like:

```js
/**
 *  
*/
const getHost = () => {
   //
   // Client side host
    if (typeof window !== 'undefined') {
      return window.location.host;
    } 

   //
   // Server side host
   // ??? How to get the server host? 
}
```

**Comment by jamesmosier** [comment]
If you want to get the request object while on the server, you would have to do it something like this on each page:

```js
export async function getServerSideProps(context) {
  const { req } = context
  return {
    props: {}, 
  }
}
```

You could alternatively use a [custom `_app.js`](https://nextjs.org/docs/advanced-features/custom-app) and get the `req` in `getInitialProps` and then pass that to your components. Although I would say the above solution is better so you could do it on a per-page basis.
